### PR TITLE
Reset all handlers. This allowing to use same server few times

### PR DESCRIPTION
**Pull request by nut-code-monkey**
E.g. for WiFi manager when hardware sepups WiFi connection and as HTTP server during hardware life cycle

### CODE REVIEW

**Comment by me-no-dev** [comment]
I would love it if you add a few words in README about that feature :)

**Comment by nut-code-monkey** [comment]
Hope last changes is more appropriate. Now method void reset() will accept AsyncCallbackWebHandler returned from .on(...). In that case user can to know anything about internal structure of Web Handler. And method could reset internal _catchAllHandler and user 'Handlers'.

**Comment by me-no-dev** [comment]
This is more of what I imagined :)
```cpp
void AsyncWebServer::reset(){
  while(_rewrites != NULL){
    AsyncWebRewrite *r = _rewrites;
    _rewrites = r->next;
    delete r;
  }
  _rewrites = NULL;
  
  while(_handlers != NULL){
    AsyncWebHandler *h = _handlers;
    _handlers = h->next;
    delete h;
  }
  _handlers = NULL;
  
  if (_catchAllHandler != NULL){
    handler->onRequest(NULL);
    handler->onUpload(NULL);
    handler->onBody(NULL);
  }
}

AsyncWebServer::~AsyncWebServer(){
  reset();
  delete _catchAllHandler;
}
```

If you want to remove handler maybe another function should be added
```cpp
bool AsyncWebServer::removeHandler(AsyncWebHandler *handler){
  if(handler == _handlers){
    _handlers = _handlers->next;
    return true;
  }
  AsyncWebHandler *h = _handlers;
  while(h != NULL){
    if(handler == h->next){
      AsyncWebHandler *d = h->next;
      h->next = d->next;
      delete d;
      return true;
    }
  }
  return false;
}
```
 Same for rewrites I guess

**Comment by me-no-dev** [comment]
Not all handlers are AsyncCallbackWebHandler :)

**Comment by me-no-dev** [comment]
BTW to reset one of the _catchAllHandler callbacks, you can simply server.onUpload(NULL) to not handle it anymore

File: src/WebServer.cpp [hunk 1]
```diff
@@ -66,6 +55,23 @@ AsyncWebRewrite& AsyncWebServer::addRewrite(AsyncWebRewrite* rewrite){
...
+    return true;
+  }
+  AsyncWebRewrite *r = _rewrites;
+  while(r != NULL){
+    if(rewrite == r->next){
+      AsyncWebRewrite *d = r->next;
+      r->next = d->next;
+      delete d;
+      return true;
+    }
```
Comment (me-no-dev) [inline src/WebServer.cpp]: need to add ```r = r->next``` or while will loop forever
same for the method below

File: README.md [hunk 2]
```diff
@@ -807,6 +807,33 @@ server.on("/scan", HTTP_GET, [](AsyncWebServerRequest *request){
 });
 ```
 
+### Remove handlers and writers
+
```
Comment (me-no-dev) [inline README.md]: Those are "rewrites" and not "writers". Please edit the README.

**Comment by me-no-dev** [comment]
Thanks :)

**Pull request merged**